error-prone-contrib/src/main/java/tech/picnic/errorprone/bugpatterns/NonStaticImport.java

```diff
@@ -0,0 +1,194 @@
+
+  /**
+   * Type members that should never be statically imported.
+   *
+   * <p>Identifiers listed by {@link #BAD_STATIC_IMPORT_CANDIDATE_IDENTIFIERS} should be omitted
+   * from this collection.
+   *
+   * <p>This should be mutually exclusive with {@link StaticImport#STATIC_IMPORT_CANDIDATE_MEMBERS}.
+   */
+  // XXX: Perhaps the set of exempted `java.util.Collections` methods is too strict. For now any
+  // method name that could be considered "too vague" or could conceivably mean something else in a
+  // specific context is left out.
+  static final ImmutableSetMultimap<String, String> BAD_STATIC_IMPORT_CANDIDATE_MEMBERS =
+      ImmutableSetMultimap.<String, String>builder()
+          .put("com.mongodb.client.model.Filters", "empty")
+          .putAll(
+              "java.util.Collections",
+              "addAll",
+              "copy",
+              "fill",
+              "list",
+              "max",
+              "min",
+              "nCopies",
+              "rotate",
+              "sort",
+              "swap")
+          .put("java.util.Locale", "ROOT")
+          .put("java.util.Optional", "empty")
+          .putAll("java.util.regex.Pattern", "compile", "matches", "quote")
+          .put("org.springframework.http.MediaType", "ALL")
+          .build();
+
+  /**
+   * Identifiers that should never be statically imported.
+   *
+   * <p>Identifiers listed by {@link StaticImport#STATIC_IMPORT_CANDIDATE_MEMBERS} should be
+   * mutually exclusive with identifiers listed here.
+   *
+   * <p>This should be a superset of the identifiers flagged by {@link
+   * com.google.errorprone.bugpatterns.BadImport}.
+   */
+  static final ImmutableSet<String> BAD_STATIC_IMPORT_CANDIDATE_IDENTIFIERS =
+      ImmutableSet.of(
+          "builder",
+          "copyOf",
+          "create",
+          "from",
+          "getDefaultInstance",
+          "INSTANCE",
+          "MAX",
+          "MAX_VALUE",
+          "MIN",
+          "MIN_VALUE",
+          "newBuilder",
+          "newInstance",
+          "of",
+          "valueOf");
+
+  /** Instantiates a new {@link NonStaticImport} instance. */
+  public NonStaticImport() {}
+
+  @Override
+  public Description matchIdentifier(IdentifierTree tree, VisitorState state) {
+    Symbol symbol = ASTHelpers.getSymbol(tree);
+    if (!isMatch(symbol, state)) {
+      return Description.NO_MATCH;
+    }
+
+    SuggestedFix.Builder builder = SuggestedFix.builder();
+    String replacement =
+        SuggestedFixes.qualifyType(state, builder, symbol.getEnclosingElement()) + ".";
+
+    return describeMatch(
+        tree,
+        builder
+            .removeStaticImport(getImportToRemove(symbol))
+            .prefixWith(tree, replacement)
+            .build());
+  }
+
+  private static String getImportToRemove(Symbol symbol) {
+    return String.join(
+        ".",
+        requireNonNull(symbol.getEnclosingElement()).getQualifiedName(),
```

Comment:
```suggestion
       symbol.owner.getQualifiedName(),
```
Now that I think of it, we can probably just use `owner` directly. That also makes me wonder if we should call it "enclosingSymbol" further down, that's probably not the correct term 🤔.